src/views/calculator.tsx

```diff
@@ -11,15 +11,30 @@ const Calculator = () => {
     result: 0,
   });
 
+  const [pastEquations, setPastEquations] = useState([<></>]);
+
   return (
     <div>
       <header className="header">
         <p>Options</p>
       </header>
-      <Display mainString={calc.displayedString} result={calc.result} />
+      <Display
+        mainString={calc.displayedString}
+        result={calc.result}
+        pastEquations={pastEquations}
+      />
       <div className="PadBtnGrid">
-        {buttons.map((btnType: string) => {
-          return <PadBtn input={btnType} calc={calc} setCalc={setCalc} />;
+        {buttons.map((btnRole: string, index) => {
```

Comment:
Since `buttons` is a `string[]` you don't need to type the iterated item `btnRole`. 

Typescript is wise to know that if it is iterating an array of strings, each item will be a string. This is called **Type by inference**. 

You can check more on [Typescript Documentation](https://www.typescriptlang.org/docs/handbook/type-inference.html) about inference types.